File: Code/sorting_algorithms/bubble_sort/BubbleSort.java
class BubbleSort{
	public static void main(String[]args){
		int[] a={67,45,789,23,46,123,56,43,90,78};
		int z=0;
		int temp=0;
		int x=0;
		while(x <= a.length){
			
			int n = 9;
			
			for(int i=1; i<=n; i++){

				if(a[z]>a[z+1]){
					temp = a[z+1];
					a[z+1] = a[z];
					a[z] = temp;
				}
				z+=1;
			}
			x++;
			c++;
			z=0;
		}

		for(int i=0; i<a.length; i++){
			System.out.println(a[i]);
		}
	}
}

File: Code/sorting_algorithms/bubble_sort/BubbleSort.py
# This is the implementation of bubble sort in python.
# The time complexity of the algorithm is O(n^2), and the space complexity is O(1).

def bubbleSort(l,arr):
	# Here, the argument l is the length of the array
	# And the argument arr is the array which has to be sorted.

	for i in range(l):
		for j in range(0,l-i-1):
			if(arr[j] > arr[j+1]):
				temp = arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = temp